### PR TITLE
Add multi-view troubleshooting guide for sentry dart

### PR DESCRIPTION
**Pull request by martinhaintz**
## DESCRIBE YOUR PR
Multi-view for web was introduced in Flutter 3.24. As the current sentry dart package is not 100% compatible with this feature I added a guide, how to use sentry with multi-view in Flutter web.

[Flutter 3.24](https://medium.com/flutter/whats-new-in-flutter-3-24-6c040f87d1e4)
[Flutter Docs for multi-view](https://docs.flutter.dev/platform-integration/web/embedding-flutter-web)

Help us prioritize incoming PRs by letting us know when the change needs to go live.
- [ ] Urgent deadline (GA date, etc.): 
- [ ] Other deadline: 
- [x] None: Not urgent, can wait up to 1 week+

## PRE-MERGE CHECKLIST

*Make sure you've checked the following before merging your changes:*

- [x] Checked Vercel preview for correctness, including links
- [ ] PR was reviewed and approved by any necessary SMEs (subject matter experts)
- [ ] PR was reviewed and approved by a member of the [Sentry docs team](https://github.com/orgs/getsentry/teams/docs)

### CODE REVIEW

File: docs/platforms/flutter/troubleshooting.mdx
Comment (buenaflor) [inline docs/platforms/flutter/troubleshooting.mdx:69]: I would not talk about a specific app like our example app but explain it more generally so users can apply it to their use cases.

This applies to the other explanations as well that you added here.

File: docs/platforms/flutter/troubleshooting.mdx
Comment (martinhaintz) [reply docs/platforms/flutter/troubleshooting.mdx:67]: If you click on the flutter docs website at [3.24.0 announcement](https://docs.flutter.dev/release/release-notes) you are getting forwarded to the medium article. So this is the official announcement.

Should I still remove the link?

Comment (buenaflor) [reply docs/platforms/flutter/troubleshooting.mdx:67]: let's keep the link to https://docs.flutter.dev/platform-integration/web/embedding-flutter-web

File: includes/troubleshooting/flutter/enable_multi_view.mdx
Comment (p-mazhnik) [inline includes/troubleshooting/flutter/enable_multi_view.mdx:208]: `flutter_bootstrap.js` will add `main.dart.js` and register `flutter_service_worker`, so these two script tags could be removed

Comment (martinhaintz) [reply includes/troubleshooting/flutter/enable_multi_view.mdx:208]: thanks for the info.

File: docs/platforms/flutter/troubleshooting.mdx
Comment (martinhaintz) [inline docs/platforms/flutter/troubleshooting.mdx:1]: @buenaflor do you have any thoughts on this? did I miss some features that are currently not supported?

File: docs/platforms/flutter/troubleshooting.mdx
Comment (buenaflor) [inline docs/platforms/flutter/troubleshooting.mdx:91]: I've been reading this a couple times and I think we shouldn't use our example app as a demonstration for several reasons

1. It has a lot of code, `main.dart` alone has 1.2k LoC which would make it even more confusing for a user.
2. It's a sandbox for us devs, while users can of course test it out we generally don't endorse someone to use our example app in the official sentry-docs, it also doesn't necessarily include any best practices, we mostly use it to validate features
3. Users would need to clone it, build and compile sentry flutter, dart and all the other packages before they can start looking into the demonstration code you've posted which are too many steps

So I would suggest instead to use a very minimal code setup (mainly `main.dart`) that anyone can copy and paste

File: docs/platforms/flutter/troubleshooting.mdx
Comment (buenaflor) [reply docs/platforms/flutter/troubleshooting.mdx:1]: I think the limitations are fine now, if users encounter more in the future we will add them here

**Comment by buenaflor** [comment]
@martinhaintz pls see my latest comment

File: docs/platforms/flutter/troubleshooting.mdx
Comment (buenaflor) [inline docs/platforms/flutter/troubleshooting.mdx:93]:
```suggestion
### Example application using the multi-view feature with Sentry Flutter
```

File: docs/platforms/flutter/troubleshooting.mdx
Comment (buenaflor) [inline docs/platforms/flutter/troubleshooting.mdx:95]:
```suggestion
```

**Review by buenaflor — COMMENTED** [review]
left 2 comments but looks good overall

File: docs/platforms/flutter/troubleshooting.mdx
Comment (buenaflor) [reply docs/platforms/flutter/troubleshooting.mdx:95]: I don't think we need this anymore, let's jump right into the guide

**Review by buenaflor — APPROVED** [review]
Let's wait for one more reviewer to review this

File: includes/troubleshooting/flutter/enable_multi_view.mdx
Comment (kahest) [inline includes/troubleshooting/flutter/enable_multi_view.mdx:10]:
```suggestion
const String exampleDsn = '___DSN___';
```

File: docs/platforms/flutter/troubleshooting.mdx
Comment (kahest) [inline docs/platforms/flutter/troubleshooting.mdx:75]: `l`: is this necessary? the above states that `WidgetsBindingIntegration` doesn't support multi-view, but it's unclear whether any of the above features need/should be deactivated

File: docs/platforms/flutter/troubleshooting.mdx
Comment (kahest) [inline docs/platforms/flutter/troubleshooting.mdx:101]:
```suggestion
Now you should be able to see **two** instances of the same application side by side, with different **ViewIds** in the `body`.
```

File: docs/platforms/flutter/troubleshooting.mdx
Comment (kahest) [inline docs/platforms/flutter/troubleshooting.mdx:93]: `m`: apologies if already discussed, but wdyt about adding this into the sentry-dart repo as sample app and pointing to that from here?

**Review by kahest — COMMENTED** [review]
left some small tweaks and clarifying questions

File: docs/platforms/flutter/troubleshooting.mdx
Comment (buenaflor) [reply docs/platforms/flutter/troubleshooting.mdx:75]: I suppose we need to be more specific, Screenshot and UserInteraction integrations are not added by default so users should not add/use them when enabling multiview but WidgetsBindingIntegration is enabled by default in the SDK which then requires the user to manually disable it.

@martinhaintz do you know if it's possible from the SDK side to detect whether multiview is enabled or not? If yes it makes things easy and we can just auto-disable unsupported features

Comment (kahest) [reply docs/platforms/flutter/troubleshooting.mdx:75]: > Screenshot and UserInteraction integrations are not added by default
ah yes, that makes sense. yeah would be good to be a bit more explicit

File: docs/platforms/flutter/troubleshooting.mdx
Comment (buenaflor) [reply docs/platforms/flutter/troubleshooting.mdx:93]: I think that's fine if the code is simple enough so the user can quickly reproduce it on their own

Comment (martinhaintz) [reply docs/platforms/flutter/troubleshooting.mdx:93]: I also think for this case the code is enough, as this feature is not that widely used and is also only relevant for web.

File: docs/platforms/flutter/troubleshooting.mdx
Comment (martinhaintz) [reply docs/platforms/flutter/troubleshooting.mdx:75]: @buenaflor @kahest  good idea 👍 I created a PR which disables `SentryScreenshotWidget`and `SentryUserInteractionWidget` for multiview web applications. https://github.com/getsentry/sentry-dart/pull/2366

File: docs/platforms/flutter/troubleshooting.mdx
Comment (lizokm) [inline docs/platforms/flutter/troubleshooting.mdx:93]:
```suggestion
### Example Application
```

File: docs/platforms/flutter/troubleshooting.mdx
Comment (lizokm) [inline docs/platforms/flutter/troubleshooting.mdx:101]:
```suggestion

Copy the `main.dart` file into the `lib` folder of your existing project. This file already contains the code of the `multi_view_app.dart` from the [`flutter documentation`](https://docs.flutter.dev/platform-integration/web/embedding-flutter-web#handling-view-changes-from-dart).
Next, copy the `flutter_bootstrap.js` file and the `index.html` file into the `web` folder.

Make sure you're using **Flutter 3.24** or newer and run the application.

Now you should be able to see **two** instances of the same application side by side, with different **ViewIds** in the `body`.
```

File: includes/troubleshooting/flutter/enable_multi_view.mdx
Comment (lizokm) [inline includes/troubleshooting/flutter/enable_multi_view.mdx:154]:
```suggestion
          If you're serving your web app in a path other than the root, change the
          href value below to reflect the base path you're serving from.

```

**Review by lizokm — APPROVED** [review]
Made a few wording suggestions, but otherwise looks good on my end.

**Comment by martinhaintz** [comment]
@buenaflor I updated the docs according to this new [PR](https://github.com/getsentry/sentry-dart/pull/2366/ )

**Comment by martinhaintz** [comment]
@lizokm I have reverted my changes because the new PR will not be merged soon. I have implemented all your changes from your last review and this PR can be merged now.
However, the CI is failing with `Vercel – changelog — Deployment has failed`.
Is there something I can do from my side, to resolve this failing CI?

**Review by buenaflor — APPROVED** [review]
I think this is ok now, we have a draft [PR](https://github.com/getsentry/sentry-dart/pull/2366) that can automatically detect if it's multiview

This won't be merged now due to lack of tests but once it is merged this doc needs to be changed or maybe even removed

**Pull request merged**
